lib/client.js

```diff
@@ -216,19 +224,20 @@ class Client extends EventEmitter {
     }
 
     const promises = [];
-    if (typeof data === 'object') {
-      const keys = Object.keys(data);
-      for (const key of keys) {
-        if (data[key] && data[key].then) {
-          promises.push(data[key]);
-          thenResolvePromisedValue(data, key);
+    if (Array.isArray(data)) {
+      for (let i = 0; i < data.length; ++i) {
+        const value = data[i];
+
+        if (value && typeof value.then === 'function') {
+          promises.push(value);
+          thenResolvePromisedValue(data, i);
         }
       }
-    } else if (data instanceof Array) {
-      for (let i = 0; i < data.length; i++) {
-        if (data[i] && data[i].then) {
-          promises.push(data[i]);
-          thenResolvePromisedValue(data, i);
+    } else if (typeof data === 'object') {
```

Comment:
An `array` is also of type `object`.

Reply:
But it's `else` section for `Array.isArray` check

Reply:
so I swapped the code blocks
first I check the data for an array type, and then for an object